cmd/ddev/cmd/composer-create.go

```diff
@@ -64,52 +63,44 @@ ddev composer create --prefer-dist --no-interaction --no-dev psr/log
 
 		composerRoot := app.GetComposerRoot(false, false)
 
-		// Make the user confirm that existing contents will be deleted
-		util.Warning("Warning: MOST EXISTING CONTENT in the Composer root (%s) will be deleted by the composer create-project operation. Only .ddev, .git and .tarballs will be preserved.", composerRoot)
-		if !composerCreateYesFlag {
-			if !util.Confirm("Would you like to continue?") {
-				util.Failed("create-project cancelled")
-			}
-		}
-
 		err = os.MkdirAll(composerRoot, 0755)
 		if err != nil {
 			util.Failed("Failed to create composerRoot: %v", err)
 		}
 
-		// Remove most contents of Composer root
-		util.Warning("Removing any existing files in Composer root")
-		objs, err := fileutil.ListFilesInDir(composerRoot)
-		if err != nil {
-			util.Failed("Failed to create project: %v", err)
-		}
-
-		for _, o := range objs {
-			// Preserve .ddev, .git, .tarballs
-			if o == ".ddev" || o == ".git" || o == ".tarballs" {
-				continue
+		appRoot := app.GetAbsAppRoot(false)
+		if appRoot != composerRoot {
+			// If composer root is not the app root, fail if it's not fully empty.
+			if !fileutil.IsDirectoryEmpty(composerRoot) {
+				util.Failed("Failed to create project: '%v' has to be empty", composerRoot)
```

Comment:
I don't understand this limitation completely. It's often the case that the composer_root is not the approot (Acquia always does it this way, composer_root: docroot). But that means that we would be creating docroot/sites/default/settings.php for example there. What is the limitation here? Fail if composer_root is not fully empty?

Reply:
Good catch, I never consider the possibility of docroot and composer_root being the same. I'll remove that and do the validation on composer_root all the time!

Reply:
this simplifies it even more!

Reply:
Fixed and tests improved!